pkg/asset/installconfig/azure/validation.go

```diff
@@ -88,59 +89,63 @@ func ValidateDiskEncryptionSet(client API, ic *types.InstallConfig) field.ErrorL
 }
 
 // ValidateInstanceType ensures the instance type has sufficient Vcpu and Memory.
-func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceType string, diskType string, req resourceRequirements, ultraSSDEnabled bool, vmNetworkingType string) field.ErrorList {
+func ValidateInstanceType(client API, fieldPath *field.Path, region, instanceType string, diskType string, req resourceRequirements, ultraSSDEnabled bool, vmNetworkingType string, capabilities map[string]string) field.ErrorList {
 	allErrs := field.ErrorList{}
-
-	typeMeta, err := client.GetVirtualMachineSku(context.TODO(), instanceType, region)
-	if err != nil {
-		return append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, err.Error()))
+	val, ok := capabilities["vCPUsAvailable"]
+	if ok {
+		cpus, err := strconv.ParseFloat(val, 0)
+		if err != nil {
+			return append(allErrs, field.InternalError(fieldPath, err))
+		}
+		if cpus < float64(req.minimumVCpus) {
+			errMsg := fmt.Sprintf("instance type does not meet minimum resource requirements of %d vCPUsAvailable", req.minimumVCpus)
+			allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
+		}
+	} else {
+		allErrs = append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, "capability not found: vCPUsAvailable"))
 	}
 
-	if typeMeta == nil {
-		errMsg := fmt.Sprintf("not found in region %s", region)
-		return append(allErrs, field.Invalid(fieldPath.Child("type"), instanceType, errMsg))
```

Comment:
This is going to break the tests which are expecting this error message for invalid instance types. The way things are now, this should be moved to `GetVMCapabilities`

Reply:
Should I also move the fieldPath.Child? That seemed out-of-place.